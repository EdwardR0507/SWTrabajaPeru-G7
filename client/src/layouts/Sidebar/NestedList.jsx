import React from "react";
import { makeStyles } from "@material-ui/core/styles";
import { Link as RouterLink } from "react-router-dom";
import Link from "@material-ui/core/Link";
import List from "@material-ui/core/List";
import ListItem from "@material-ui/core/ListItem";
import ListItemIcon from "@material-ui/core/ListItemIcon";
import ListItemText from "@material-ui/core/ListItemText";
import IconButton from "@material-ui/core/IconButton";
import Collapse from "@material-ui/core/Collapse";
import HomeIcon from "@material-ui/icons/Home";
import PersonIcon from "@material-ui/icons/Person";
import WorkIcon from "@material-ui/icons/Work";
import AddIcon from "@material-ui/icons/Add";
import SettingsIcon from "@material-ui/icons/Settings";

const useStyles = makeStyles((theme) => ({
  root: {
    width: "100%",
    maxWidth: 360,
    backgroundColor: theme.palette.background.paper,
  },
  nested: {
    paddingLeft: theme.spacing(4),
  },
}));

export default function NestedList() {
  const classes = useStyles();
  const [open, setOpen] = React.useState(true);

  const handleClick = () => {
    setOpen(!open);
  };

  return (
    <List
      component="nav"
      aria-labelledby="nested-list-subheader"
      className={classes.root}
    >
      <ListItem button>
        <ListItemIcon>
          <HomeIcon />
        </ListItemIcon>
        <ListItemText primary="Inicio" />
      </ListItem>
      <ListItem button>
        <ListItemIcon>
          <PersonIcon />
        </ListItemIcon>
        <ListItemText primary="Cliente" />
      </ListItem>
      <ListItem button onClick={handleClick}>
        <ListItemIcon>
          <WorkIcon />
        </ListItemIcon>
        <ListItemText primary="Trabajador" />
      </ListItem>
      <Collapse in={open} timeout="auto" unmountOnExit>
        <List component="div" disablePadding>
          <ListItem button className={classes.nested}>
            <ListItemIcon>
              <AddIcon />
            </ListItemIcon>
            <Link component={RouterLink} to="/newservice">
              <ListItemText primary="Registrar Servicios" />
            </Link>
          </ListItem>
          <ListItem button className={classes.nested}>
            <ListItemIcon>
              <SettingsIcon />
            </ListItemIcon>
            <Link component={RouterLink} to="/manageservices">
              <ListItemText primary="Mis Servicios" />
            </Link>
          </ListItem>
        </List>
      </Collapse>
    </List>
  );
}
