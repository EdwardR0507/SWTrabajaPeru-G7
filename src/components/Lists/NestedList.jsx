import React from "react";
import { useHistory, useLocation } from "react-router";
import { makeStyles } from "@material-ui/core/styles";
import List from "@material-ui/core/List";
import ListItem from "@material-ui/core/ListItem";
import ListItemIcon from "@material-ui/core/ListItemIcon";
import ListItemText from "@material-ui/core/ListItemText";
import Collapse from "@material-ui/core/Collapse";
import HomeIcon from "@material-ui/icons/Home";
import PersonIcon from "@material-ui/icons/Person";
import WorkIcon from "@material-ui/icons/Work";
import SettingsIcon from "@material-ui/icons/Settings";

const useStyles = makeStyles((theme) => ({
  root: {
    width: "100%",
    maxWidth: 360,
    backgroundColor: theme.palette.background.paper,
  },
  nested: {
    paddingLeft: theme.spacing(4),
  },
}));

export default function NestedList() {
  const classes = useStyles();
  const [worker, setWorker] = React.useState(false);
  const [client, setClient] = React.useState(false);
  const location = useLocation();
  const history = useHistory();
  const state = location.state;

  const handleClick = () => {
    setWorker(!worker);
  };

  const handleOpen = () => {
    setClient(!client);
  };

  return (
    <List
      component="nav"
      aria-labelledby="nested-list-subheader"
      className={classes.root}
    >
      <ListItem
        button
        onClick={() => {
          history.push({
            pathname: "/",
            state: { token: state?.token },
          });
        }}
      >
        <ListItemIcon>
          <HomeIcon />
        </ListItemIcon>
        <ListItemText primary="Inicio" />
      </ListItem>
      <ListItem button onClick={handleOpen}>
        <ListItemIcon>
          <PersonIcon />
        </ListItemIcon>
        <ListItemText primary="Cliente" />
      </ListItem>
      <Collapse in={client} timeout="auto" unmountOnExit>
        <List component="div" disablePadding>
          <ListItem
            button
            className={classes.nested}
            onClick={() => {
              history.push({
                pathname: "/hiredServices",
                state: { token: state.token },
              });
            }}
          >
            <ListItemIcon>
              <SettingsIcon />
            </ListItemIcon>
            <ListItemText primary="Mis Servicios Contratados" />
          </ListItem>
        </List>
      </Collapse>
      <ListItem button onClick={handleClick}>
        <ListItemIcon>
          <WorkIcon />
        </ListItemIcon>
        <ListItemText primary="Trabajador" />
      </ListItem>
      <Collapse in={worker} timeout="auto" unmountOnExit>
        <List component="div" disablePadding>
          <ListItem
            button
            className={classes.nested}
            onClick={() => {
              history.push({
                pathname: "/manageservices",
                state: { token: state.token },
              });
            }}
          >
            <ListItemIcon>
              <SettingsIcon />
            </ListItemIcon>
            <ListItemText primary="Mis Servicios" />
          </ListItem>
          <ListItem
            button
            className={classes.nested}
            onClick={() => {
              history.push({
                pathname: "/solicitedServices",
                state: { token: state.token },
              });
            }}
          >
            <ListItemIcon>
              <SettingsIcon />
            </ListItemIcon>
            <ListItemText primary="MIS SOLICITUDES" />
          </ListItem>
        </List>
      </Collapse>
    </List>
  );
}
