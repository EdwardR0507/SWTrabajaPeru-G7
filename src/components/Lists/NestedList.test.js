/*Importamos las librerias principales*/
import React from "react";
import '@testing-library/jest-dom/extend-expect';
import { shallow } from "enzyme";
import NestedList from "./NestedList";
import { Router } from "react-router-dom";
import { createMemoryHistory } from 'history';
/*Renderizado del componente NestedList*/
describe("NestedList", () => {
  it('render "NestedList"', () => {
    const history = createMemoryHistory()
    const wrapper = shallow(<Router history={history}>
      <NestedList />
    </Router>);
    expect(wrapper).toMatchSnapshot() 
  });
  it("test handleClick", () => {
    const history = createMemoryHistory()
    const wrapper = shallow(<Router history={history}>
      <NestedList />
    </Router>);
    const button = wrapper.find(NestedList).shallow().find({prop: "button"});
    const open = wrapper.find(NestedList).shallow().find({prop: "in"});
    button.simulate("click")
    expect(open.props("in")).toBe(true)
  })
});
