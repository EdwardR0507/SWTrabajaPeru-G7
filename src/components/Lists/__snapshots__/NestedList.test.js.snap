// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NestedList render "NestedList" 1`] = `ShallowWrapper {}`;
